libdeno/binding.cc

```diff
@@ -201,12 +206,43 @@ void Print(const v8::FunctionCallbackInfo<v8::Value>& args) {
       args.Length() >= 2 ? args[1]->BooleanValue(context).ToChecked() : false;
   bool prints_newline =
       args.Length() >= 3 ? args[2]->BooleanValue(context).ToChecked() : true;
+#ifdef _WIN32
+  HANDLE std_handle =
+      GetStdHandle(is_err ? STD_ERROR_HANDLE : STD_OUTPUT_HANDLE);
+
+  // does not have associated standard handles,
+  // or has not redirected to a associated standard handles
+  if (std_handle == nullptr || std_handle == INVALID_HANDLE_VALUE) {
+    return;
+  }
+
+  DWORD mode;
+  // `WriteConsole` only works with console screen, not files nor pipes.
+  // if the handle is redirected to a file, use `WriteFile`
+  if (!GetConsoleMode(std_handle, &mode)) {
+    WriteFile(std_handle, *str, str.length() * sizeof(**str), nullptr, nullptr);
+  } else {
+    const int n = MultiByteToWideChar(CP_UTF8, 0, *str, -1, nullptr, 0);
+    wchar_t* wbuf = (wchar_t*)malloc(n * sizeof(wchar_t));
+    MultiByteToWideChar(CP_UTF8, 0, *str, -1, wbuf, n);
```

Comment:
Can you capture the return value from `MultiByteToWideChar` and assert that it succeeded?
I know theoretically it should fit because you just obtained the buffer size --iff nobody made a mistake about trailing \0 (not) being counted etc.